src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -48,6 +48,9 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
     {
         if (!i_offset)
         {
+            if (i_target->IsWithinDistInMap(owner, 1.0f))
```

Comment:
This will fail for large models - it makes more sense to check against bounding box.